appinventor/components/src/com/google/appinventor/components/runtime/ChartData2D.java

```diff
@@ -68,16 +68,30 @@ public void AddEntry(final String x, final String y) {
       threadRunner.execute(new Runnable() {
         @Override
         public void run() {
+          YailList pair = YailList.makeList(Arrays.asList(x, y));
+          YailList labelList = container.Labels();
+          int index = labelList.indexOf(x);
+
           try {
             // Create a 2-tuple, and add the tuple to the Data Series
-            YailList pair = YailList.makeList(Arrays.asList(x, y));
             dataModel.addEntryFromTuple(pair);
+          } catch (NumberFormatException e){
+            // likely that the x value is a string/label, get the corresponding index and use that as x
+            YailList pairLabelAsIndex = YailList.makeList(Arrays.asList(index, y));
+            dataModel.addEntryFromTuple(pairLabelAsIndex);
+            // should we do another catch here?
 
-            // Refresh Chart with new data
-            onDataChange();
+            /*catch( NumberFormatException nfe)
+              this.view.getForm().dispatchErrorOccurredEvent(this.view.chartComponent,
+            "GetEntryFromTuple",
+            ErrorMessages.ERROR_INVALID_CHART_ENTRY_VALUES,
+            rawX, rawY);
+
+             */
```

Comment:
@ewpatton do you think we should do another catch here or is it sufficient to assume that accessing the index of the label will be sufficient?